packages/playwright/src/runner/watchMode.ts

```diff
@@ -134,21 +139,27 @@ export async function runWatchModeLoop(configLocation: ConfigLocation, initialOp
   let result: FullResult['status'] = 'passed';
 
   while (true) {
-    printPrompt();
-    const readCommandPromise = readCommand();
-    await Promise.race([
+    if (bufferMode)
+      printBufferPrompt(dirtyTestFiles, teleSuiteUpdater.config!.rootDir);
+    else
+      printPrompt();
+
+    const command = await Promise.race([
       onDirtyTests,
-      readCommandPromise,
+      readCommand(),
     ]);
-    if (!readCommandPromise.isDone())
-      readCommandPromise.resolve('changed');
 
-    const command = await readCommandPromise;
+    if (bufferMode && command === 'changed')
+      continue;
+
+    const commandToTriggerTestRun = (bufferMode ? 'run' : 'changed');
```

Comment:
I think something like `shouldRunChangedFiles: boolean` will be easier to understand. I had to wrap my head around this construct.